src/spring-cloud/azext_spring_cloud/_app_validator.py

```diff
@@ -8,8 +8,8 @@
 from azure.cli.core.azclierror import InvalidArgumentValueError
 from msrestazure.azure_exceptions import CloudError
 from azure.core.exceptions import (ResourceNotFoundError)
-from ._client_factory import cf_spring_cloud
 from ._resource_quantity import (validate_cpu as validate_cpu_value, validate_memory as validate_memory_value)
+from ._client_factory import cf_spring_cloud_20220101preview
```

Comment:
May I ask why did you add the api-version number to the name of `_client_factory` ?

Reply:
Here follows the current implementation about client factory. The default client factory uses the latest stable API version. But some features in previewed is only available in preview API version. Ideally, we, service team will flush all preview features in a new stable API version, at that time, we can remove the `factory_preview_api_version` and replace them as the new default client factory.

Reply:
Got it, thanks